### PR TITLE
add a nyquist factor to OscilGen::get

### PR DESCRIPTION
**Pull request by friedolino78**
implemented as ordered ;-)

### CODE REVIEW

File: src/Synth/OscilGen.cpp [hunk 1]
```diff
@@ -1059,18 +1059,21 @@ short int OscilGen::get(float *smps, float freqHz, int resonance)
...
+        nyquistLimited = synth.oscilsize / 2;
 
     //Process harmonics
     {
         int realnyquist = nyquist;
 
         if(Padaptiveharmonics != 0)
             nyquist = synth.oscilsize / 2;
-        for(int i = 1; i < nyquist - 1; ++i)
+        for(int i = 1; i < nyquistLimited - 1; ++i)
```
Comment (JohannesLorenz) [inline src/Synth/OscilGen.cpp:1076]: `Padaptiveharmonics` has no effect anymore now.

Comment (friedolino78) [reply src/Synth/OscilGen.cpp:1076]: > `Padaptiveharmonics` has no effect anymore now.

it is also used inside OscilGen::adaptiveharmonic()
look at line 913

Comment (JohannesLorenz) [reply src/Synth/OscilGen.cpp:1076]: What I really meant was that the 2 lines above this line do not have an effect on this for loop anymore. Which means that this code breaks the current behavior.

**Comment by JohannesLorenz** [comment]
I remember that I asked for this because we need AA for the wavetable branch, but is this really the right tool for that job? (Didn't you even write a PR which does the AA?)

**Comment by friedolino78** [comment]
> I remember that I asked for this because we need AA for the wavetable branch, but is this really the right tool for that job? (Didn't you even write a PR which does the AA?)

yes, this one can be dumped.

**Pull request closed without merging**